### PR TITLE
Feature/quizzes list on angularjs

### PR DESCRIPTION
**Pull request by superNazo**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=484&view=planning&selectedIssue=LR-180)

### Author.
Nazarii Ivankiv

- [x] I have added link to this PR to the JIRA
- [x] I have logged time in JIRA.

### Reviewers.

- [x] @anatoliiibrahimov
- [x] @OKrain
- [x] @vovka

### Issue summary.

Rework quizzes list page on angular js

### Fix summary. 

Created quizzes list page on angular js, added search filter, and filters by id/name/date, created pagination for 5 items per page

### Testing approach.
There are unit tests for indexQuiz controller. The are testing sort function using jasmine framework

### CODE REVIEW

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 1]
```diff
@@ -52,50 +52,47 @@ describe("Bind Cocoon events to link", function(){
...
-    it("Add new input to the form", function() {
-      var $scope = {};
-      $controller("newQuizCtrl", {$scope: $scope});
+  describe("indexQuizCtrl", function(){
 
-      $scope.add();
+    describe("$scope.sort", function(){
+      it("should change the sorting key", function(){
+        var $scope = {};
+        $controller("indexQuizCtrl", {$scope: $scope});
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["$controller" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2432630352/issues/source?bid=3205144&fileBranchId=3241901#l66)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 2]
```diff
@@ -52,50 +52,47 @@ describe("Bind Cocoon events to link", function(){
...
 
-  describe("$scope.sort", function(){
-    it("should change the sorting key", function(){
-      var $scope = {};
-      $controller("indexQuizCtrl", {$scope: $scope});
+        $scope.sort();
 
-      $scope.sort("name");
-      expect($scope.sortKey).toEqual("name");
+        expect($scope.reverse).toBe(true);
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["expect" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2432630352/issues/source?bid=3205144&fileBranchId=3241901#l80)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js [hunk 3]
```diff
@@ -52,50 +52,47 @@ describe("Bind Cocoon events to link", function(){
   });
 });
 
-describe("newQuizCtrl", function() {
-  beforeEach(module("quizzesControllers"));
+describe("Controllers tests", function() {
```
Comment (vovka) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: Codacy Issue found: ["describe" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2432630352/issues/source?bid=3205144&fileBranchId=3241901#l55)

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 1 above)
Comment (superNazo) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it is defined in angularjs framework

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 2 above)
Comment (superNazo) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it is defined in jasmine testing framework

File: spec/javascripts/angularjs_app/quizzes_controller_spec.js (hunk 3 above)
Comment (superNazo) [inline spec/javascripts/angularjs_app/quizzes_controller_spec.js]: it is defined in jasmine testing framework

**Comment by vovka** [comment]
🆗

**Comment by vovka** [comment]
Broken link to jira.

**Comment by superNazo** [comment]
fixed

**Comment by OKrain** [comment]
good :+1:

**Pull request merged**